script.js

```diff
@@ -1,4 +1,419 @@
+// makes all of the deck of cards, arrange by rank, suits and name
+
+var makeDeck = function () {
+  var cardDeck = [];
+
+  var suitIndex = 0;
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    console.log("Current Suit : " + currentSuit);
+
+    // rank 1-13
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suits: currentSuit,
+        rank: rankCounter,
+      };
+      console.log("rank : " + rankCounter);
+
+      cardDeck.push(card);
+      rankCounter = rankCounter + 1;
+    }
+    suitIndex = suitIndex + 1;
+  }
+  return cardDeck;
+};
+
+// generate the function from above into variable so easy to use
+var cardDeck = makeDeck();
+
+// helps to get all the random numbers
+var getRandomNumberIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// pulls the random number as an randomindex and search into the card deck for the number
+var shuffleCards = function (cardDeck) {
+  currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomNumberIndex(cardDeck.length);
+    var randomCardGenerated = cardDeck[randomIndex]; // this takes the random index which is a random number and then it will look for that number on the cards deck.
+    var currentCardGenerated = cardDeck[currentIndex]; // using the current index it will find a card in the deck
+    // it swaps random card and current card together for a result
+    cardDeck[currentIndex] = randomCardGenerated;
+    cardDeck[randomIndex] = currentCardGenerated;
+
+    currentIndex = currentIndex + 1;
+  }
+
+  return cardDeck;
+};
+
+// store the user name
+var username = [];
+var gameMode = "enter name";
+
+var checkUsername = function (input) {
+  gameMode == "enter name";
+  if (input != ``) {
+    gameMode = "start game";
+
+    username = input;
+    console.log("Check username input:", username);
+    var message = "Hi 😀" + username + " <br> </br> Click submit to play.🃏";
+  } else if (input == ``) {
+    gameMode == "enter name";
+    message = "Please enter your name 👀.";
+    return message;
+  }
+  return message;
+};
+
+// store the player scores in global variable from functions below
+var playerOneTwoCardResult = [];
```

Comment:
consider renaming this variable to playerScore? also, does it have to be an array? could it just be a number?